src/app/modules/form/public-form/__tests__/public-form.controller.spec.ts

```diff
@@ -389,4 +421,741 @@ describe('public-form.controller', () => {
+        expect(MockAuthService.getFormIfPublic).toHaveBeenCalledWith(
+          MOCK_FORM_ID,
+        )
+        // 2. Check that error is correct
+        expect(
+          MockFormService.checkFormSubmissionLimitAndDeactivateForm,
+        ).not.toHaveBeenCalled()
+        expect(MOCK_RES.status).toHaveBeenCalledWith(404)
+      })
+
+      it('should return 404 when the form is private and not accessible by the public', async () => {
+        // Arrange
+        // 1. Mock the response
+        const MOCK_RES = expressHandler.mockResponse()
+
+        // 2. Mock the call to retrieve the form
+        MockAuthService.getFormIfPublic.mockReturnValueOnce(
+          errAsync(
+            new PrivateFormError('teehee this form is private', 'private form'),
+          ),
+        )
+
+        // Act
+        await PublicFormController.handleGetPublicForm(
+          MOCK_REQ,
+          MOCK_RES,
+          jest.fn(),
+        )
+
+        // Assert
+        // 1. Check args of mocked services
+        expect(MockAuthService.getFormIfPublic).toHaveBeenCalledWith(
+          MOCK_FORM_ID,
+        )
+        // 2. Check that error is correct
+        expect(
+          MockFormService.checkFormSubmissionLimitAndDeactivateForm,
+        ).not.toHaveBeenCalled()
+        expect(MOCK_RES.status).toHaveBeenCalledWith(404)
+      })
+    })
+
+    describe('errors in form access', () => {
+      const MOCK_JWT_PAYLOAD: JwtPayload = {
+        userName: 'mock',
+        rememberMe: false,
+      }
+
+      beforeAll(() => {
+        MockFormService.setMyInfoError.mockImplementation((publicForm) =>
+          ok({ ...publicForm, myInfoError: false }),
+        )
+      })
+
+      it('should return 200 with isIntranetUser set to false when a user accesses the form with no authType set', async () => {
+        // Arrange
+        const MOCK_NIL_AUTH_FORM = (mocked({
+          ...BASE_FORM,
+          authType: AuthType.NIL,
+        }) as unknown) as MockedObject<IPopulatedForm>
+        const MOCK_RES = expressHandler.mockResponse()
+
+        MockAuthService.getFormIfPublic.mockReturnValueOnce(
+          okAsync(MOCK_NIL_AUTH_FORM),
+        )
+        MockFormService.checkFormSubmissionLimitAndDeactivateForm.mockReturnValueOnce(
+          okAsync(MOCK_NIL_AUTH_FORM),
+        )
+
+        MockFormService.setIsIntranetFormAccess.mockImplementation(
+          (_, publicForm) => ok({ ...publicForm, isIntranetUser: false }),
+        )
+
+        // Act
+        await PublicFormController.handleGetPublicForm(
+          MOCK_REQ,
+          MOCK_RES,
+          jest.fn(),
+        )
+
+        // Assert
+        expect(MOCK_RES.json).toHaveBeenCalledWith({
+          form: MOCK_NIL_AUTH_FORM.getPublicView(),
+          isIntranetUser: false,
+          myInfoError: false,
+        })
+      })
+
+      it('should return 200 with isIntranetUser set to true when a user accesses the form using SP', async () => {
```

Comment:
should we update the test description to

`should return 200 with isIntranetUser set to true when a user accesses an Authtype.SP form in Intranet`